src/learner.cc

```diff
@@ -300,13 +308,49 @@ class LearnerImpl : public Learner {
     for (auto& p_metric : metrics_) {
       p_metric->Configure({cfg_.begin(), cfg_.end()});
     }
+  }
 
+  void Load(dmlc::Stream* fi) override {
+    monitor_.Start(__func__);
+    if (cfg_.find("model_format") != cfg_.cend()) {
+      tparam_.model_format = cfg_.at("model_format");
+    }
+    if (tparam_.model_format == "json") {
+      LOG(WARNING) << "JSON serialization is at experimental stage. "
+                   << "Output schema is subject to change in the future.";
+      auto  ReadStream =
+          [](dmlc::Stream* fi) {
+            std::string buffer;
```

Comment:
I suggest not to use strings as arrays of bytes.
As far as I remember C++ standard only for std::vector guarantees that
std::vector x;
&x[0] would actually point to a consecutive array of x.size() elements

while most implementation of std::string do the same I would still advice to use std::vector<char> given that we do not use any string specific functions anyway

Reply:
@trams See https://stackoverflow.com/questions/1986966/does-s0-point-to-contiguous-characters-in-a-stdstring .  Using string eliminates a copy and plays nice with `dmlc::Stream`.  Should be safe here, not nice looking though.